master/internal/kubernetes/request_queue.go

```diff
@@ -98,6 +100,9 @@ type requestQueue struct {
 	queue                    []*queuedResourceRequest
 	pendingResourceCreations map[*actor.Ref]*queuedResourceRequest
 	availableWorkers         []*actor.Ref
+
+	creationInProgress       map[*actor.Ref]bool
```

Comment:
blocking: it looks like the approach in this PR adds some unnecessary complexity. Can we merge `creationInProgress` with `pendingResourceCreations` to be like below and remove `blockedResourceDeletions`?

```
// resourceCreations keeps track of both pending and in-progress creation requests. If the value is true, it's already sent to a worker to process.
resourceCreations map[queueResourceRequest]bool
```

Reply:
I do agree this is more complex than necessary right now. I tried refactoring a bit but didn't end up with good result. Do you mind if we land this in the current state and come back to this later in the week (when its not as time sensitive).

Reply:
Since this is a low probability case that users would hit, I'd just write a final solution to save us time.

Reply:
I'd really prefer not to ship known bugs ...